compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/VerifyCompilerInputLegality.cpp

```diff
@@ -37,31 +37,51 @@ struct VerifyCompilerStableHloInputLegality final
     // NOTE: It is not fully illegal to tunnel input dialect ops through to
     // backends that expect them. When such situations arise, the container
     // op should be marked recursively legal here.
-    SmallVector<Diagnostic> failures;
+    SmallVector<Diagnostic> failuresDiagnostics;
     {
-      ScopedDiagnosticHandler diag(context,
-                                   [&](Diagnostic &d) -> LogicalResult {
-                                     failures.push_back(std::move(d));
-                                     return success();
-                                   });
+      ScopedDiagnosticHandler diag(
+          context, [&](Diagnostic &d) -> LogicalResult {
+            failuresDiagnostics.push_back(std::move(d));
+            return success();
+          });
       if (succeeded(applyPartialConversion(getOperation(), conversionTarget,
                                            std::move(conversionPatterns)))) {
-        return;
+        // Continue checking for other possible errors.
       }
     }
 
+    // Also check for dialects that should have been converted to stablehlo
+    // prior to being presented to IREE in the first place.
+    // We don't want to use source dependencies for these dialects, so we just
+    // use string matching.
+    llvm::DenseSet<StringRef> invalidDialects{"tf", "mhlo"};
+    llvm::DenseSet<Operation *> invalidOps;
+    getOperation().walk([&](Operation *op) {
+      if (invalidDialects.contains(op->getName().getDialectNamespace())) {
```

Comment:
This is doing string comparisons, I'd rather check the IDs.

Reply:
Are the IDs available without a source dep on the TF / MHLO dialects?

See the comment above:
```
    // We don't want to use source dependencies for these dialects, so we just
    // use string matching.
```